metadatastore/commands.py

```diff
@@ -839,6 +841,11 @@ def _transform_data(data, timestamps):
     to storage format:
         {'data': {<key>: (<value>, <timestamp>)}.
     """
+    # shallow copy so we can do some norrmalization
+    data = dict(data)
```

Comment:
I do not think so as without this PR if you try to put an np.ndarray into mongo it dies because it does not know how to encode the ndarray into json.   

I do not think we want to leave the model where a timestamp is a single scalar.  If you want to put a time series into an event and the exact timestamps are important then that should go in as a data key.